src/components/SidePanel/SidePanel.jsx

```diff
@@ -0,0 +1,178 @@
+ * @param {Array} props.selectedGenres - Array of selected genres.
+ * @param {function} props.setAvailabilities - Function to set availabilities.
+ * @param {Array} props.availabilities - Array of availabilities.
+ * @param {function} props.setRelease - Function to set release.
+ * @param {Array} props.release - Array of release types.
+ * @returns {JSX.Element} The SidePanel component.
+ */
+
+export default function SidePanel({
+  getMovies,
+  setSelectedGenres,
+  selectedGenres,
+  setAvailabilities,
+  availabilities,
+  setRelease,
+  release,
+  setSelectedKeyword,
+  selectedKeyword,
+}) {
+  const [isActiveFilter, setIsActiveFilter] = useState(false);
+  const [lang, setLang] = useState([]);
+  const [country, setCountry] = useState([]);
+  const [genres, setGenres] = useState([]);
+
+  /**
+   * Fetches movie genres from the API and sets them using the provided callback.
+   */
+  const getGenres = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/genre/movie/list?language=en&api_key=${apiKey}`;
+    axios
+      .get(apiUrl)
+      .then((response) => {
+        setGenres(response.data.genres);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  /**
+   * Fetches available languages from the API and sets them using the provided callback.
+   */
+  const getLanguages = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/configuration/languages?api_key=${apiKey}`;
+    axios.get(apiUrl).then((response) => {
+      setLang(response.data);
+    });
+  };
+
+  /**
+   * Fetches available countries from the API and sets them using the provided callback.
+   */
+  const getCountry = () => {
+    const apiKey = "4db3b4ee5893cead9657d41699ec4c26";
+    const apiUrl = `https://api.themoviedb.org/3/configuration/countries?language=en-US&api_key=${apiKey}`;
+    axios.get(apiUrl).then((response) => {
+      setCountry(response.data);
+    });
+  };
+
+  /**
+   * Toggles the display of the content filter and fetches genres, languages, and countries.
+   * @param {object} e - Event object.
+   */
+  const showContentFilter = (e) => {
+    e.preventDefault();
+    setIsActiveFilter((prevIsActive) => !prevIsActive);
+    getGenres();
+    getLanguages();
+    getCountry();
+  };
+
+  return (
+    <div>
+      <form onSubmit={getMovies}>
+        <SortInput />
+        <ContentContainer>
+          <div style={{ display: "flex", justifyContent: "space-between" }}>
+            <ContentName onClick={showContentFilter}>Filter</ContentName>
+            <Icon src={arrow} alt="arrow" isActive={isActiveFilter} />
+          </div>
+
+          <div
+            style={{
+              display: isActiveFilter ? "block" : "none",
+              padding: "0 0.625rem",
+            }}
+          >
+            <Paragraph>Show Me</Paragraph>
```

Comment:
Suggestion: can we rename the `Paragraph` styled component to `Text`? I believe this makes more sense, especially since this is being used as a label or as a prefix many times.